src/cache.rs

```diff
@@ -11,13 +11,7 @@ pub type ResourcePathBuf = PathBuf;
 
 pub trait ResourceCache {
```

Comment:
Templates should be cached in the resource cache. Feel free to add any methods neccesary to the trait (Just keep them consistent with the tileset ones)